test/images/vulnerability-scan/scan-image.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Pipeline Service Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+
+VULNERABILITIES_EXIST=1
+
+get_digest() {
+  #number of retries=3
+  for _ in {1..3}
+  do
+    #Get the digest of the image
+    digest_http_code=$(curl -s -w '%{http_code}' -o /tmp/digest.json -H "Accept: application/vnd.quay.distribution.manifest.v2+json Content-type: application/json Authorization: Bearer $AUTH_BEARER_TOKEN" https://quay.io/api/v1/repository/redhat-pipeline-service/"$IMAGE_NAME")
+
+    if [[ "$digest_http_code" == "200" ]]; then
+      digest=$(jq '.tags[] | select(.name == "main").manifest_digest ' /tmp/digest.json | tr -d '"')
+      rm -f /tmp/digest.json
+      get_vulnerabilities
+      break
+    else
+      printf "Error while fetching digests from Quay. Status code: %s\n" "${digest_http_code}"
+    fi
+  done
+}
+
+get_vulnerabilities() {
+  #Scan for vulnerabilities
+  results_http_code=$(curl -s -w '%{http_code}' -o /tmp/vulnerability.json -H "Content-type: application/json Authorization: Bearer $AUTH_BEARER_TOKEN" https://quay.io/api/v1/repository/redhat-pipeline-service/"$IMAGE_NAME"/manifest/"$digest"/security?vulnerabilities=true)
+
+  if [[ "$results_http_code" == "200" ]]; then
+    printf "\n *********************************** \n"
+    printf " \t %s results " "$IMAGE_NAME"
+    printf "\n *********************************** \n"
+
+    #Filtering results with vulnerabilities only.
+    results=$(jq -r '.data.Layer.Features[].Vulnerabilities[]' /tmp/vulnerability.json)
+
+    if [[ -n $results ]]; then
+      printf "%s" "$results" | jq
+      printf "\n Check the full report here: https://quay.io/repository/redhat-pipeline-service/%s/manifest/%s?tab=vulnerabilities \n" "$IMAGE_NAME" "$digest"
+    else
+      printf "No vulnerabilities found \n"
+      VULNERABILITIES_EXIST=0
+    fi
+
+    rm -f /tmp/vulnerability.json
```

Comment:
Just wondering whether we need that. I don't see where it would play a role later on in the script and my understanding is that there is no persistent storage mounted, meaning that it would get deleted when the container exits anyway.

Reply:
It may be different if the cod is shared with what is used for multiple scans.

Reply:
Sure, I just added that to keep it clean. I don't think it matters either way because like you said, the container will die once its done. Deleted the line.